src/router/BaseRouter.php

```diff
@@ -27,22 +27,20 @@ abstract class BaseRouter<+TResponder> {
       $data = Dict\map($data, $value ==> \urldecode($value));
       return tuple($responder, new ImmMap($data));
     } catch (NotFoundException $e) {
-      foreach (HttpMethod::getValues() as $next) {
-        if ($next === $method) {
-          continue;
-        }
-        try {
-          list($responder, $data) = $resolver->resolve($next, $path);
-          if ($method === HttpMethod::HEAD && $next === HttpMethod::GET) {
-            $data = Dict\map($data, $value ==> \urldecode($value));
-            return tuple($responder, new ImmMap($data));
-          }
-          throw new MethodNotAllowedException();
-        } catch (NotFoundException $_) {
-          continue;
-        }
+      $allowed = $this->getAllowedMethods($path);
+      if (0 === C\count($allowed)) {
+        throw $e;
       }
-      throw $e;
+
+      if (
+        HttpMethod::HEAD === $method && keyset[HttpMethod::GET] === $allowed
```

Comment:
ditto for yoda  conditions

Reply:
![image](https://user-images.githubusercontent.com/29315886/69190514-d9410f00-0b20-11ea-99da-9d86c45d96f7.png)

Reply:
fwiw, assignment-as-an-expression should be dead soon and a type error; @lexidor 's been fixing a bunch of them